hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hss/signschedule/SignScheduleTranslatorTest.java

```diff
@@ -352,4 +393,225 @@ void testGasRequirement() {
+        given(enhancement.nativeOperations()).willReturn(nativeOperations);
+        given(nativeOperations.getSchedule(anyLong())).willReturn(schedule);
+        given(schedule.scheduleId()).willReturn(scheduleID);
+
+        attempt = prepareHssAttemptWithBytesAndCustomConfig(
+                Bytes.wrapByteBuffer(
+                        SignScheduleTranslator.AUTHORIZE_SCHEDULE.encodeCall(Tuple.of(APPROVED_HEADLONG_ADDRESS))),
+                subject,
+                enhancement,
+                addressIdConverter,
+                verificationStrategies,
+                signatureVerifier,
+                gasCalculator,
+                configuration);
+
+        final var returnedScheduleId = subject.scheduleIdFor(attempt);
+        assertThat(returnedScheduleId).isEqualTo(scheduleID);
+    }
+
+    @Test
+    void testScheduleIdForScheduleService() {
+        given(enhancement.nativeOperations()).willReturn(nativeOperations);
+        given(nativeOperations.getSchedule(anyLong())).willReturn(schedule);
+        given(schedule.scheduleId()).willReturn(scheduleID);
+
+        attempt = prepareHssAttemptWithBytesAndCustomConfig(
+                Bytes.wrapByteBuffer(SignScheduleTranslator.SIGN_SCHEDULE.encodeCall(
+                        Tuple.of(APPROVED_HEADLONG_ADDRESS, new byte[0]))),
+                subject,
+                enhancement,
+                addressIdConverter,
+                verificationStrategies,
+                signatureVerifier,
+                gasCalculator,
+                configuration);
+
+        final var returnedScheduleId = subject.scheduleIdFor(attempt);
+        assertThat(returnedScheduleId).isEqualTo(scheduleID);
+    }
+
+    @Test
+    void testGetKeysForSignScheduleWhenVerified() {
+        given(enhancement.nativeOperations()).willReturn(nativeOperations);
+        given(nativeOperations.getSchedule(anyLong())).willReturn(schedule);
+        given(schedule.scheduleId()).willReturn(scheduleID);
+        given(configuration.getConfigData(ContractsConfig.class)).willReturn(contractsConfig);
+        given(contractsConfig.chainId()).willReturn(296);
+        given(signatureVerifier.verifySignature(any(), any(), any(), any(), any()))
+                .willReturn(true);
+
+        final var sigMapBytes = getSigMapKnownKeyTypeBytes(296);
+        attempt = prepareHssAttemptWithBytesAndCustomConfig(
+                Bytes.wrapByteBuffer(SignScheduleTranslator.SIGN_SCHEDULE.encodeCall(
+                        Tuple.of(APPROVED_HEADLONG_ADDRESS, sigMapBytes.toByteArray()))),
+                subject,
+                enhancement,
+                addressIdConverter,
+                verificationStrategies,
+                signatureVerifier,
+                gasCalculator,
+                configuration);
+
+        final var keySet = SignScheduleTranslator.getKeyForSignSchedule(attempt);
+
+        assertThat(keySet).contains(ecdsaKey).contains(ed25519);
```

Comment:
FYI, in AssertJ `assertThat(keySet).containsOnly(ecdsaKey, ed25519);` is not only a few characters shorter it also would confirm that no other key snuck into the key set.